components/raftstore/src/store/peer.rs

```diff
@@ -1259,6 +1261,9 @@ where
                 "region_id" => self.region_id,
                 "peer_id" => self.peer.get_id(),
             );
+            if let Some(snap_ctx) = self.apply_snap_ctx.as_mut() {
+                snap_ctx.destroy_peer_after_apply = true;
```

Comment:
why not just set `self.apply_snap_ctx = None`

Reply:
Setting it to None has much bigger impact and error prone, as apply_snap_ctx None or not is critical in many places. 
It breaks the original behavior that apply_snap_ctx is only set when the apply is actually finished or aborted (not cancelling). 
The way of this PR does not impact any original behavior. What we need is a hint so that when apply finished, we redo the destroy peer.